zone/zone.cpp

```diff
@@ -2043,100 +2043,168 @@ bool ZoneDatabase::LoadStaticZonePoints(LinkedList<ZonePoint *> *zone_point_list
+					iterator.GetData()->GetZ(),
+					iterator.GetData()->GetHeading()
+				).c_str()
+			);
+
+			if (time_remaining != 0xFFFFFFFF) {
+				auto seconds_remaining = (time_remaining / 1000);
+				client->Message(
+					Chat::White,
+					fmt::format(
+						"Spawn {} | Respawn: {} ({} Second{})",
+						spawn_number,
+						ConvertSecondsToTime(seconds_remaining),
+						seconds_remaining,
+						seconds_remaining != 1 ? "s" : ""
+					).c_str()
+				);
+			}
+
+			filtered_count++;
+			spawn_number++;
 		}
 
-		x++;
+		spawn_count++;
+
 		iterator.Advance();
 	}
 
-	client->Message(Chat::White, "%i of %i spawns listed.", iEnabledCount, x);
-}
-
-void Zone::ShowDisabledSpawnStatus(Mob* client)
-{
-	LinkedListIterator<Spawn2*> iterator(spawn2_list);
-	int x = 0;
-	int iDisabledCount = 0;
-
-	iterator.Reset();
-
-	while(iterator.MoreElements())
-	{
-		if (iterator.GetData()->timer.GetRemainingTime() == 0xFFFFFFFF)
-		{
-			client->Message(Chat::White, "  %d: %1.1f, %1.1f, %1.1f: disabled", iterator.GetData()->GetID(), iterator.GetData()->GetX(), iterator.GetData()->GetY(), iterator.GetData()->GetZ());
-			iDisabledCount++;
-		}
-
-		x++;
-		iterator.Advance();
+	if (!spawn_count) {
+		client->Message(Chat::White, "No spawns were found in this zone.");
+		return;
 	}
 
-	client->Message(Chat::White, "%i of %i spawns listed.", iDisabledCount, x);
+	if (!filtered_count) {
+		client->Message(
+			Chat::White,
+			fmt::format(
+				"No {} spawns were found in this zone.",
+				status_type
+			).c_str()
+		);
+		return;
+	}
+	
+	if (status_type == "All") {
+		client->Message(
+			Chat::White,
+			fmt::format(
+				"{} Spawn{} listed.",
+				spawn_count,
+				spawn_count != 1 ? "s" : ""
+			).c_str()
+		);
+	} else {
+		client->Message(
+			Chat::White,
+			fmt::format(
+				"{} Of {} spawn{} listed.",
+				filtered_count,
+				spawn_count,
+				spawn_count != 1 ? "s" : ""
+			).c_str()
+		);
+	}
 }
 
-void Zone::ShowSpawnStatusByID(Mob* client, uint32 spawnid)
+void Zone::ShowSpawnStatusByID(Client *client, uint32 spawn_id)
 {
-	LinkedListIterator<Spawn2*> iterator(spawn2_list);
-	int x = 0;
-	int iSpawnIDCount = 0;
+	uint32 found_count = 0;
+	uint32 spawn_count = 0;
 
+	LinkedListIterator<Spawn2*> iterator(spawn2_list);
 	iterator.Reset();
 
-	while(iterator.MoreElements())
-	{
-		if (iterator.GetData()->GetID() == spawnid)
-		{
-			if (iterator.GetData()->timer.GetRemainingTime() == 0xFFFFFFFF)
-				client->Message(Chat::White, "  %d: %1.1f, %1.1f, %1.1f: disabled", iterator.GetData()->GetID(), iterator.GetData()->GetX(), iterator.GetData()->GetY(), iterator.GetData()->GetZ());
-			else
-				client->Message(Chat::White, "  %d: %1.1f, %1.1f, %1.1f: %1.2f", iterator.GetData()->GetID(), iterator.GetData()->GetX(), iterator.GetData()->GetY(), iterator.GetData()->GetZ(), (float)iterator.GetData()->timer.GetRemainingTime() / 1000);
-
-			iSpawnIDCount++;
+	while (iterator.MoreElements()) {
+		if (iterator.GetData()->GetID() == spawn_id) {
```

Comment:
Could shorten things up a bit by putting just below the while

```cpp
auto &e = iterator.GetData();
```

Then `e->GetID()` etc.

Reply:
Done.